sdk/src/extras/disk.rs

```diff
@@ -107,7 +108,12 @@ pub mod builder {
             impl Future<Output = Result<(), DiskImportError>> + 'a,
             Error<crate::types::Error>,
         > {
-            Ok(self.execute_with_control()?.0)
+            let (fut, handle) = self.execute_with_control()?;
+
+            // Leak the handle so we don't cancel the request by dropping it.
+            mem::forget(handle);
```

Comment:
this seems a bit inelegant. Did we consider other options such as a `run_without_cancel` function, ignoring cancelation from `cancel_rx` in `run_with_cancel` or returning a new `Future` here that holds onto the handle so that it isn't dropped.

Reply:
That's fair, I've added a `run` method that doesn't take a cancel channel.